components/src/components/Map/Component.ts

```diff
@@ -0,0 +1,181 @@
+import { Components } from 'formiojs';
+const FieldComponent = (Components as any).components.field;
+import MapService from './services/MapService';
+import baseEditForm from './Component.form';
+import * as L from 'leaflet';
+
+const DEFAULT_CENTER: [number, number] = [
+  53.96717190097409, -123.98320425388914,
+]; // Ensure CENTER is a tuple with exactly two elements
+
+export default class Component extends (FieldComponent as any) {
+  static schema(...extend) {
+    return FieldComponent.schema({
+      type: 'map',
+      label: 'Map',
+      key: 'map',
+      input: true,
+      defaultvalue: { features: [] },
+      ...extend,
+    });
+  }
+
+  static get builderInfo() {
+    return {
+      title: 'Map',
+      group: 'basic',
+      icon: 'map',
+      weight: 70,
+      schema: Component.schema(),
+    };
+  }
+
+  static editForm = baseEditForm;
+
+  componentID: string;
+  mapService: MapService;
+
+  constructor(component, options, data) {
+    super(component, options, data);
+    this.componentID = super.elementInfo().component.id;
+  }
+
+  render() {
+    return super.render(
+      `<div id="map-${this.componentID}" style="height:400px; z-index:1;"> </div>`
```

Comment:
can we make the height a property the designer sets?

Reply:
Yes, but height customizability never really came up in the ux reviews. I can pass it by minshen

Reply:
After talking to minshen, it doesn't seem like a high priority for designers to choose this. It's possible we can take a look at an advanced tab with some of these settings down the road.